teapots/common/ndk_helper/CMakeLists.txt

```diff
@@ -1,16 +1,33 @@
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
 
-add_library(ndk-helper STATIC
-            gestureDetector.cpp
-            gl3stub.cpp
-            GLContext.cpp
-            interpolator.cpp
-            JNIHelper.cpp
-            perfMonitor.cpp
-            sensorManager.cpp
-            shader.cpp
-            tapCamera.cpp
-            vecmath.cpp)
+add_library(NdkHelper
+  STATIC
+    gestureDetector.cpp
+    gl3stub.cpp
+    GLContext.cpp
+    interpolator.cpp
+    JNIHelper.cpp
+    perfMonitor.cpp
+    sensorManager.cpp
+    shader.cpp
+    tapCamera.cpp
+    vecmath.cpp
+)
 
-target_include_directories(ndk-helper PRIVATE
-                           ${ANDROID_NDK}/sources/android/native_app_glue)
+set_target_properties(NdkHelper
+  PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+)
+target_compile_features(NdkHelper
+  PRIVATE cxx_noexcept
+)
+target_compile_options(NdkHelper
+  PRIVATE
+    -Wall -Werror -fno-rtti
+)
+
+target_include_directories(NdkHelper
+  PRIVATE
+    ${ANDROID_NDK}/sources/android/native_app_glue
+)
```

Comment:
If you change the above three to PUBLIC from private, then you can eliminate all the other places that rely upon NdkHelper.  Additionally, if you have a common include folder, you can take advantage of interfaces like this:
set_target_properties( NdkHelper PROPERTIES INTERFACE_INCLUDE_DIRECTORIES <commonPath> )

Also, add this:
target_link_libraries(NdkHelper cpufeatures native_app_glue android log)
Now whenever you use NdkHelper in the parent target_link_libraries lines, you don't need any of them.
Finally, native_app_glue doesn't need to be specifically called out here at all; you can use:
include(AndroidNdkModules)
android_ndk_import_module_native_app_glue()
then the native_app_glue target lib is defined.